### PR TITLE
Added fmax and fmin operators

### PR DESCRIPTION
**Pull request by lijiaqi0612**
### PR types
New features

### PR changes
 OPs 

### Describe
Added fmax and fmin operators, and added documentation and unit tests for comparison with competing products.

example:

```
import numpy as np
import paddle

x = paddle.to_tensor([2, 3, np.nan], dtype='float32')
y = paddle.to_tensor([1, np.nan, 5], dtype='float32')

# fmax
res_fmax = paddle.fmax(x, y)
print(res_fmax)
#    [ 2., 3., 5.]
 
# fmin
res_fmin = paddle.fmin(x, y)
print(res_fmin)
#    [ 1., 3., 5.]
```

The reason for the need to modify the 'no_grad_set_white_list.py' file is that because there are two inputs, the reverse of one of them needs to be tested separately, and the operator name needs to be added to the whitelist.

### CODE REVIEW

File: paddle/fluid/operators/elementwise/elementwise_max_op.cc [hunk 1]
```diff
@@ -103,3 +120,32 @@ REGISTER_OP_VERSION(elementwise_max)
...
+    ops::ElementwiseFMaxKernel<paddle::platform::CPUDeviceContext, int>,
+    ops::ElementwiseFMaxKernel<paddle::platform::CPUDeviceContext, int64_t>);
+REGISTER_OP_CPU_KERNEL(
+    elementwise_fmax_grad,
+    ops::ElementwiseFMaxGradKernel<paddle::platform::CPUDeviceContext, float>,
+    ops::ElementwiseFMaxGradKernel<paddle::platform::CPUDeviceContext, double>,
+    ops::ElementwiseFMaxGradKernel<paddle::platform::CPUDeviceContext, int>,
+    ops::ElementwiseFMaxGradKernel<paddle::platform::CPUDeviceContext, int64_t>);
+
+REGISTER_OP_VERSION(elementwise_fmax)
```
Comment (iclementine) [inline paddle/fluid/operators/elementwise/elementwise_max_op.cc]: 这一段不需要，这是用于 op 兼容性升级的。

File: python/paddle/tensor/math.py [hunk 2]
```diff
@@ -631,6 +631,126 @@ def minimum(x, y, name=None):
             x, y, axis=axis, act=act, op_name=op_type)
     return _elementwise_op(LayerHelper(op_type, **locals()))
 
+def fmax(x, y, name=None):
+    """
+    Compares the elements at the corresponding positions of the two tensors and returns a new tensor containing the maximum value of the element.
+    The equation is:
+
+    .. math::
+        out = fmax(x, y)
```
Comment (iclementine) [inline python/paddle/tensor/math.py:641]: 文档里面最好写明 nan 的传播机制。亦即 nan 被认为小于任何值。

File: python/paddle/tensor/math.py [hunk 3]
```diff
@@ -631,6 +631,126 @@ def minimum(x, y, name=None):
...
+def fmin(x, y, name=None):
+    """
+    Compares the elements at the corresponding positions of the two tensors and returns a new tensor containing the minimum value of the element.
+    The equation is:
+
+    .. math::
+        out = fmin(x, y)
+
+    **Note**:
+    ``paddle.fmin`` supports broadcasting. If you want know more about broadcasting, please refer to :ref:`user_guide_broadcasting` .
```
Comment (iclementine) [inline python/paddle/tensor/math.py:705]: 文档里面最好写明 nan 的传播机制。亦即 nan 被认为大于任何值。

File: paddle/fluid/operators/elementwise/elementwise_max_op.cu [hunk 4]
```diff
@@ -56,3 +56,16 @@ REGISTER_OP_CUDA_KERNEL(
     ops::ElementwiseMaxGradKernel<paddle::platform::CUDADeviceContext, int>,
     ops::ElementwiseMaxGradKernel<paddle::platform::CUDADeviceContext,
                                   int64_t>);
+
+REGISTER_OP_CUDA_KERNEL(
+    elementwise_fmax,
```
Comment (iclementine) [inline paddle/fluid/operators/elementwise/elementwise_max_op.cu:61]: float16 和 bfloat16 支持一下？

File: python/paddle/fluid/tests/unittests/test_fmax_op.py [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+
+from __future__ import print_function
+
+import unittest
+import numpy as np
+import paddle
+import paddle.fluid.core as core
+
+
+class ApiFMaxTest(unittest.TestCase):
```
Comment (iclementine) [inline python/paddle/fluid/tests/unittests/test_fmax_op.py:24]: 还缺少 OpTest。
需要继承 OpTest 写 Op 单元测试。可以参考 python/paddle/fluid/tests/unittests/test_elementwise_add_op.py

fmin 也是一样

Comment (lijiaqi0612) [reply python/paddle/fluid/tests/unittests/test_fmax_op.py:24]: ok 已经添加了

File: python/paddle/tensor/math.py (hunk 3 above)
Comment (lijiaqi0612) [reply python/paddle/tensor/math.py:705]: OK 已经添加了

Comment (lijiaqi0612) [reply python/paddle/tensor/math.py:705]: OK 已经添加了

File: paddle/fluid/operators/elementwise/elementwise_max_op.cu (hunk 4 above)
Comment (lijiaqi0612) [reply paddle/fluid/operators/elementwise/elementwise_max_op.cu:61]: OK  已经支持了fp16，框架里elementwise体系不支持bfloat16

File: python/paddle/tensor/__init__.py [hunk 6]
```diff
@@ -295,6 +297,8 @@
            'maximum',
            'min',
            'minimum',
+           'fmax',
+           'fmin',
```
Comment (dingjiaweiww) [inline python/paddle/tensor/__init__.py:311]: 这里需要把函数加入此文件的tensor_method_func列表

File: python/paddle/__init__.py [hunk 7]
```diff
@@ -437,6 +439,8 @@
            'pow',
            'zeros_like',
            'maximum',
+           'fmax',
+           'fmin',
```
Comment (dingjiaweiww) [inline python/paddle/__init__.py]: 需要加入此文件的all 列表

**Review by dingjiaweiww — DISMISSED** [review]

File: python/paddle/__init__.py (hunk 7 above)
Comment (lijiaqi0612) [reply python/paddle/__init__.py]: 已经加了呀

File: paddle/fluid/operators/elementwise/elementwise_max_op.cc [hunk 8]
```diff
@@ -53,6 +53,25 @@ class ElementwiseMaxOpMaker : public ElementwiseOpMaker {
...
+    AddInput("X", "The first tensor holding the elements to be compared.");
+  }
+
+  void AddInputY() override {
+    AddInput("Y", "The second tensor holding the elements to be compared.");
+  }
+
+  std::string GetOpFuntionality() const override {
+    return "Compare two tensors and returns a new tensor containing the "
+           "element-wise maxima.";
```
Comment (jeff41404) [inline paddle/fluid/operators/elementwise/elementwise_max_op.cc]: what's different between fmax and max, shall we describe it more detailed？

File: paddle/fluid/operators/elementwise/elementwise_min_op.cc [hunk 9]
```diff
@@ -53,6 +53,25 @@ class ElementwiseMinOpMaker : public ElementwiseOpMaker {
...
+    AddInput("X", "The first tensor holding the elements to be compared.");
+  }
+
+  void AddInputY() override {
+    AddInput("Y", "The second tensor holding the elements to be compared.");
+  }
+
+  std::string GetOpFuntionality() const override {
+    return "Compare two tensors and returns a new tensor containing the "
+           "element-wise minima.";
```
Comment (jeff41404) [inline paddle/fluid/operators/elementwise/elementwise_min_op.cc]: what's different between fmin and min, shall we describe it more detailed？

Comment (lijiaqi0612) [reply paddle/fluid/operators/elementwise/elementwise_min_op.cc]: OK

**Review by jeff41404 — DISMISSED** [review]
lgtm

**Review by lanxianghit — DISMISSED** [review]

**Review by Shixiaowei02 — DISMISSED** [review]
approve for no_grad_set_white_list

**Review by lanxianghit — APPROVED** [review]

**Review by Shixiaowei02 — APPROVED** [review]

**Review by dingjiaweiww — APPROVED** [review]

**Review by jeff41404 — APPROVED** [review]

**Review by iclementine — APPROVED** [review]
LGTM

**Pull request merged**